fv_be/app/models/dictionary.py

```diff
@@ -0,0 +1,125 @@
+    def __str__(self):
+        return self.text
+
+
+class Acknowledgment(BaseModel):
+    """Model for acknowledgments associated to each dictionary entry."""
+
+    # from fv:acknowledgments, fv:source, fv:reference, fv-word:acknowledgement, fv-phrase:acknowledgement
+    text = models.TextField()
+
+    def __str__(self):
+        return self.text
+
+
+class Translation(BaseModel):
+    """Model for translations associated to each dictionary entry."""
+
+    # Choices for Language
+    ENGLISH_ENUM_KEY = "EN"
+    FRENCH_ENUM_KEY = "FR"
+    LANGUAGE_CHOICES = [
+        (ENGLISH_ENUM_KEY, _("English")),
+        (FRENCH_ENUM_KEY, _("French")),
+    ]
+
+    # Fields
+    translation = models.CharField(max_length=200)
+    language = models.CharField(
+        max_length=2, choices=LANGUAGE_CHOICES, default=ENGLISH_ENUM_KEY
+    )
+    # from fv-word:part_of_speech
+    part_of_speech = models.ManyToManyField(PartOfSpeech)
+    # todo: more representative name for the following attribute ?
+    parent = models.ForeignKey("DictionaryEntry", on_delete=models.CASCADE)
+
+    def __str__(self):
+        return _("Translation in %(language)s: %(translation)s.") % {
+            "language": self.language,
+            "translation": self.translation,
+        }
+
+
+class AlternateSpelling(BaseModel):
+    """Model for alternate spellings associated to each dictionary entry."""
+
+    # todo: more representative name for the following attribute ?
+    # from fv:alternate_spelling, fv-word:alternate_spellings, fv-phrase:alternate_spellings
+    text = models.CharField(max_length=200)
+    # todo: more representative name for the following attribute ?
+    parent = models.ForeignKey("DictionaryEntry", on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.text
+
+
+class Pronunciation(BaseModel):
+    """Model for pronunciations associated to each dictionary entry."""
+
+    # todo: more representative name for the following attribute ?
+    # from fv-word:pronunciation
+    text = models.CharField(max_length=200)
+    # todo: more representative name for the following attribute ?
+    parent = models.ForeignKey("DictionaryEntry", on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.text
+
+
+class DictionaryEntry(BaseModel):
+    """Model for dictionary entries"""
+
+    # Choices for Type
+    WORD_ENUM_KEY = "WORD"
+    PHRASE_ENUM_KEY = "PHRASE"
+    TYPE_OF_ENTRY_CHOICES = [(WORD_ENUM_KEY, _("Word")), (PHRASE_ENUM_KEY, _("Phrase"))]
+
+    # Fields
+    # from dc:title
+    title = models.CharField(max_length=200)
+    type = models.CharField(
+        max_length=6, choices=TYPE_OF_ENTRY_CHOICES, default=WORD_ENUM_KEY
+    )
+    # todo: Link to site model when available
+    # may be inherited from an abstract base class or mixin later
+    # site = models.ForeignKey(
+    #     'Site',
+    #     on_delete=models.CASCADE
+    # )
+    notes = models.ForeignKey("Note", on_delete=models.SET_NULL, null=True)
```

Comment:
The V2 UI currently supports 1 Entry with multiple Notes, so it should be Notes that have the ForeignKey to the entry. I see that the dictionary model draft suggests M:M for this relation but I don't think this makes sense. Even if multiple entries have the same Note text, the actual Note should not be shared across different entries -- a user wouldn't expect editing the Note on one entry to affect the Note on another entry. -- Happy to discuss this more if needed.